hail/Makefile

```diff
@@ -41,13 +42,25 @@ SCALA_BUILD_INFO := src/main/resources/build-info.properties
 
 SHADOW_JAR := build/libs/hail-all-spark.jar
 SHADOW_TEST_JAR := build/libs/hail-all-spark-test.jar
+DEBUG_CLASSES := build/debug
 PYTHON_JAR := python/hail/backend/hail-all-spark.jar
 WHEEL := build/deploy/dist/hail-$(HAIL_PIP_VERSION)-py3-none-any.whl
 EGG := build/deploy/dist/hail-$(HAIL_PIP_VERSION)-py3.6.egg
 
 .PHONY: shadowJar
+ifdef HAIL_DEBUG_MODE
+shadowJar: debugShadowJar
+endif
 shadowJar: $(SHADOW_JAR)
 
+.PHONY: debugShadowJar
+debugShadowJar: $(SHADOW_JAR)
```

Comment:
I don't think we should be adding another phony target like this. The shadow jar rule should just add the debug classes.